src/sdk/namespace_fs.js

```diff
@@ -1332,6 +1341,37 @@ class NamespaceFS {
         };
     }
 
+    async _get_version_path(params, fs_context) {
+        const file_path = this._get_file_path(params);
+        if (params.version_id) {
+            const stat = await nb_native().fs.stat(fs_context, file_path);
+            if (isDirectory(stat)) throw Object.assign(new Error('NoSuchKey'), { code: 'ENOENT' });
```

Comment:
I ran some examples on AWS and noticed that head object with version-id='1' throws BadRequest and not ENOENT (as the current implementation will throw), so we need here a check for an invalid version parameter.
It should be a helper function so it can be used in all the functions